drivers/clock_control/Kconfig

```diff
@@ -9,6 +9,13 @@
 #
 # Clock controller drivers
 #
+
+config CLOCK_CONTROL_MUST_USE_OUT_OF_TREE_DRIVER
```

Comment:
Maybe this could be called `CLOCK_CONTROL_OUT_OF_TREE_DRIVER` or the like, to shorten it a bit. It might be clear anyway that there must be an out-of-tree driver if it's on.

Reply:
(@carlescufi) We had a naming discussion here: https://github.com/zephyrproject-rtos/zephyr/pull/13631#discussion_r260318620

The reason for the `_MUST_USE_` (originally `_HAS_`)  was to keep consistency with other select-able Kconfig symbols. If we keep select-able symbols on the form `<noun>_<verb>_<noun>` (`FOO_HAS_BAR`), it is much easier to keep your sanity when working with the Kconfig files.

The name is verbose, but I'd rather sacrifice verbosity for clarity.

Reply:
The problem with OUT_OF_TREE_DRIVER is that we need some option/way to distinguish between the scenario where there exists an out-of-tree driver that may be selected, and the scenario where there exists an out-of-tree driver that must be selected.

And OUT_OF_TREE_DRIVER does not reflect this.

Reply:
If you really want to enforce the `<noun>_<verb>_<noun>` (which I don't think we could call a convention right now except for the `_HAS_` case) then I would go for something like `*_SELECT_OOT` or `REQUIRE_OOT` something like that. `MUST_USE` sounds convoluted and verbose to me.

Reply:
The original proposal of _HAS_ didn't make sense, but now I'm unsure if we actually considered using an option without HAS. As Ulf says, if the OOT driver is enabled, that does imply that the other drivers can't be. The symbol for an enabled out-of-tree driver would look like:

CLOCK_CONTROL_OUT_OF_TREE_DRIVER